### ISSUE TITLE
The client can't communicate with the server

### ISSUE THREAD

**Issue by yunxiao3** [mccorby/PhotoLabeller#5]
Hello, why do I use your code, the client and the server can't communicate? I checked that the phone did not upload images to the server after taking a photo.

**Comment by jacorbachosky** [comment]
Have you checked the IP of your server?
Did the phone get a new model from the server?

**Comment by yunxiao3** [comment]
When the app starts, it will show model loaded but it seems that there is no new model. I can use the browser to access serverip:9997/service/federatedservice/available on the phone to see yes. But when I use the app, I use wireshark to show that no packets are sent to the server. Thanks

**Comment by JieZhou0805** [comment]
How could I know whether the model is the renewed?

**Comment by VatsaPatel** [comment]
Have you put model.zip in your model directory?

**Comment by mccorby** [comment]
The server IP must also be set in the app. Have you done that?
It is set in `gradle.properties`

```
parameter_server_url=http://192.168.0.33:9997/service/federatedservice/
```

**Comment by VatsaPatel** [comment]
Also model will be incompactible if dl4j versions are different on client and server.

**Comment by yunxiao3** [comment]
I configured the app's parameter_server_url, but when i click the save button it does not respond and the settings button will flash back. Whether the app and my phone version are not adapted causes these problems. Is it necessary to save the data first to transfer the data to the client?

**Comment by VatsaPatel** [comment]
Yes, if you dont have model.zip initially in your  directorory, you will get exception and server will stop.

**Comment by JieZhou0805** [comment]
What is the model directory?

**Comment by JieZhou0805** [comment]
I couldn't find the model directory in the code of client.

**Comment by vito3** [comment]
I can see model.zip but should I put it in that directory? Do you need to extract it?

**Comment by yunxiao3** [comment]
I can see model.zip but should I put it in what directory? Do you need to extract it?

**Comment by JieZhou0805** [comment]
Got it. It’s very kind of you to send it to me.

Best Regards,

--

JieZhou
Student @ NCU

________________________________
发件人: Yunxiao Du <notifications@github.com>
发送时间: Thursday, April 4, 2019 4:28:05 PM
收件人: mccorby/PhotoLabeller
抄送: JieZhou0805; Comment
主题: Re: [mccorby/PhotoLabeller] The client can't communicate with the server (#5)


Closed #5<https://github.com/mccorby/PhotoLabeller/issues/5>.

―
